ruby/ql/src/queries/meta/TaintSteps.ql

```diff
@@ -0,0 +1,17 @@
+/**
+ * @name Taint steps
+ * @description The number of default taint steps.
+ * @kind metric
+ * @metricType project
+ * @metricAggregate sum
+ * @tags meta
+ * @id rb/meta/taint-steps
+ */
+
+import ruby
+import internal.TaintMetrics
+import codeql.ruby.dataflow.internal.TaintTrackingPublic
+
+predicate relevantStep(DataFlow::Node pred, DataFlow::Node succ) { localTaintStep(pred, succ) }
+
+select projectRoot(), count(DataFlow::Node pred, DataFlow::Node succ | relevantStep(pred, succ))
```

Comment:
This query kind worked in `dist-compare`, but I don't think DCA ever had support for it. Meta queries have to be `@kind problem`.

Reply:
And that had too many results and gave OOMs. 
So now I'm counting taint-steps pr. file.

Reply:
I think it's (unfortunately) normal for meta-queries to OOM the report-generator if run as a standalone query, the question is whether the _diff_ is large enough to cause an OOM in practice.

Reply:
The baseline is empty in [this new evaluation](https://github.com/github/codeql-dca-main/tree/data/erik-krogh/pr-10704-c1fae9__nightly__code-scanning/reports), and that didn't OOM the report-generator on `nightly.yml`, so that should be good enough.

Reply:
My concern wasn't that this particular query would OOM, but rather that it isn't working when used on a diff, and it's better to have one that only works on a diff than one that doesn't.

When selecting the steps directly, like in `CallGraph.ql`
- **Can not** generate report when run without a baseline (sadly)
- **Can** see source locations for individual changes
- **Can** see aggregate number of changes at the top of the report
- **Can** see changes when an equal number of results were gained and lost

When selecting the number as part of the alert message:
- **Can** generate report when run without a baseline (although in the [linked evaluation](https://github.com/github/codeql-dca-main/tree/data/erik-krogh/pr-10704-c1fae9__nightly__code-scanning/reports) it doesn't seem to render the alert message for some reason)
- **Can not** see source locations for individual changes
- **Can not** see aggregate number of changes at the top of the report (have to manually subtract numbers from the alert messages)
- **Can not** see changes when an equal number of results were gained and lost

Reply:
You're right, the `taint-steps` meta query will probably be confusing to interpret results from as it currently is.  
I don't see how we could fix that, so I'll remove it for now.